### DISCUSSION TITLE
Add option for default case insensitive

### DISCUSSION THREAD

**Discussion by DanAvni** [datalust/seq-tickets#1456] [Ideas]
Please add an option (global or user level) to work by default in case insensitive mode and have a 'cs' operator for case sensitive if needed. 99% of my searches are case insensitive and I prefer not to write 'ci' all the time (if I forget to I get strange results)

**Comment by nblumhardt** [comment]
Thanks for the suggestion, we'll give this some thought.

**Comment by nblumhardt** [comment]
Hi @DanAvni - thanks again for the suggestion. Are the searches you're referring to ones that you manually type in, e.g. `Environment = 'test'`, or do you want CI behavior from pre-written ones like the "green tick" Find menu options?

We're not keen on a global change to equality semantics (Seq relies on its own query language internally in many places, and generated/shared code would work unreliably if equality semantics vary between users/installs), but we're open to finding ways to make the UI friendlier for CI use.

Other kinds of options we can explore are "parameterized" saved queries (such as "Find machine" parameterized by machine name, for example).

**Comment by DanAvni** [comment]
Hi @nblumhardt, actually I use both the search and signals and in both cases I failed because of something:

I had a case that for a certain API key the Environment was "Test" and for another key it was "TEST". when my signal was created some events were not listed and it took me a long while to figure it out because when reading the API keys extra attributes I was reading "Test" and "TEST" as the same thing (I do not think in case sensitive)

Another case is some inconsistency in my experience when searching for some text. Continuing the previous example, imagine I have an event with Env=Test and {Env} is written in the message. If you search for "Test" or "TEST" or "test" you will always find the event because it is case insensitive. However, if you search for Env='test' you will not find the event (unless you hit the correct casing or use the ci operator). 
The same happens with GUIDs we have from various places: .NET renders GUID as lower case but SQL server as upper case. That means I cannot copy the GUID from SQL server and search for GUID=’B2053524-B89D-4AEB-AAC9-02629AF2DF58’ because that would not yield any events.

if it was up to me, I would reverse the logic completely making every search case insensitive unless specifically specified by a cs operator.

**Comment by nblumhardt** [comment]
Thanks for the feedback, Dani. Unfortunately, there are a lot of cases (haha) in which case matters, too; we've had to pick an option and it's now somewhat fixed - but there are some more interesting possibilities to explore.

Would it improve your experience if we were able to show some kind of indicator based on whether or not an expression was case-sensitive? (I.e. below the search box, where we show "Text" today? Or inline, somehow?)

The idea of providing a button/helper to insert `ci` into simple searches is also intriguing.

I suspect this will turn out to be a fairly complex space to experiment in, but keen to see what kinds of options we might have, anyway :-)

**Comment by DanAvni** [comment]
I think the options of adding some kind of toggle on the UI might be useful (so I can say if my search is ci or cs). another option could be to re-write my query by inserting ci after each string comparison. But both these options seem to be inferior to the solution of querying with ci as a default.

maybe some "AutoComplete" of ci every time I write a string?

Only programmers think in case sensitive way and even then I miss from time to time seeing a casing difference. Most other users (I think) are looking for a text. They do not care how it is written in the log (lower case/upper case/camel case/pascal case/other). That is why I think the whole case sensitivity should be flipped completely. only when I want to search for something which is case sensitive, I will specify it explicitly. In other cases it will case insensitive (or the bigger alternative of supporting the mode of operation ci or cs). While I do understand the implications, I do think it is the best option to go with is flipping the usage completely but if you want to maintain backwards compatability I think mode of operation selection should be added.

**Comment by nblumhardt** [comment]
Thanks again for this suggestion; we're very much of the opinion that we can improve usability around this, and we now have some internal proposals going for consideration later this year. We'll move this to a discussion in the meantime, as we try to keep the issues list to untriaged or committed work as much as possible.
